llvm/test/CodeGen/X86/bfloat.ll

```diff
@@ -22,10 +22,7 @@ define void @add(ptr %pa, ptr %pb, ptr %pc) nounwind {
 ; X86-NEXT:    vaddss %xmm0, %xmm1, %xmm0
 ; X86-NEXT:    vmovss %xmm0, (%esp)
 ; X86-NEXT:    calll __truncsfbf2
-; X86-NEXT:    fstps {{[0-9]+}}(%esp)
-; X86-NEXT:    vmovd {{.*#+}} xmm0 = mem[0],zero,zero,zero
-; X86-NEXT:    vmovd %xmm0, %eax
-; X86-NEXT:    movw %ax, (%esi)
+; X86-NEXT:    vmovsh %xmm0, (%esi)
```

Comment:
This change seems to will miss the truncate operation.

Reply:
What's the mean by truncate?

Reply:
Sorry being not clear. I saw the original codes containing `; X86-NEXT: movw %ax, (%esi)` seems like we are changing from a word store to a xmm store here.

Reply:
`vmovsh` can store the low 16-bit to memory directly.
The original codes has ABI mistake, which store `bf16` without `f32`. Since `f32` uses X87 registers on 32-bit target, it need to store to another memory first, reload to store again.
The patch makes the result in XMM0, so one `vmovsh` is enough.

Reply:
Thanks for explaining.